spec/roomtypes.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spec
+
+const (
+	// Content key for room type in m.room.create event https://spec.matrix.org/v1.7/client-server-api/#mroomcreate
+	MRoomCreateTypeContentKey = "type"
```

Comment:
This should probably just go into the `CreateContent` struct at https://github.com/matrix-org/gomatrixserverlib/blob/main/eventcontent.go#L29-L47